README.md

```diff
@@ -0,0 +1,195 @@
+## Creating the DUH document
+NOTE: this tutorial was made using version 1.15.0 of DUH
+
+The DUH document is a JSON5 file that describes an IP block similar to IPXACT.
+The DUH JSON schema is defined [here](https://github.com/sifive/duh-schema).
+DUH is also a suite of tools that help users author DUH documents and generate
+useful artifacts from them. This section describes how to use some of the DUH
+tools to author a DUH document for the PIO block.
+
+Subsections:
+* [Installation](#installing-duh)
+  - how to install DUH
+* [Initialization](#initializing-the-duh-document)
+  - how to initialize a DUH document
+* [Ports](#importing-verilog-ports)
+  - how to automatically import port definitions into DUH from verilog source
+* [Bus Interfaces](#define-bus-interfaces)
+  - how to automatically infer bus definitions
+* [Memory Maps](#define-memory-maps)
+  - how to define memory maps in DUH
+* [Parameter Schema](#define-parameter-schema)
+  - how to define a parameter schema in DUH
+
+### Installing DUH
+First you will need to install [DUH](https://github.com/sifive/duh). To install
+DUH in your current directory run
+```bash
+npm i duh@1.15.0
+```
+
+This will create a `node_modules` subdirectory in your current directory with
+all the package sources. The `duh` executable along with other useful tools are
+installed in `node_modules/.bin`.
+
+### Initializing the DUH document
+To create an initial DUH document run `duh init` and answer the prompts. For
+this tutorial we will name the block `pio` and write the DUH document to
+`block-pio-sifive/pio.json5`
+<pre>
+duh init
+? <b>Document file name</b> pio.json5
+? <b>Block name</b> pio
+? <b>version</b> 0.1.0
+? <b>Please write a short description about the block</b> A Parallel IO block
+? <b>Block type</b> component
+? <b>Source type</b> Verilog
+</pre>
+
+### Importing verilog ports
+Fill in the port definitions of the block in the DUH document. The
+`duh-import-verilog-ports` tool can parse the verilog and fill in the
+definitions for you.
+```bash
+cat rtl/pio/pio.sv | duh-import-verilog-ports pio.json5
+```
+
+### Define bus interfaces
```

Comment:
Here is a separate project to practice on bus inference procedures: https://github.com/sifive/block-ark/blob/master/README.md

Reply:
added link